taglib/riff/wav/wavproperties.h

```diff
@@ -52,35 +52,107 @@ namespace TagLib {
         /*!
          * Create an instance of WAV::Properties with the data read from the
          * ByteVector \a data.
+         *
+         * \deprecated
          */
         Properties(const ByteVector &data, ReadStyle style);
 
         /*!
          * Create an instance of WAV::Properties with the data read from the
          * ByteVector \a data and the length calculated using \a streamLength.
+         *
+         * \deprecated
          */
         Properties(const ByteVector &data, uint streamLength, ReadStyle style);
 
         /*!
+         * Create an instance of WAV::Properties with the data read from the
+         * ByteVector \a data and the length calculated using \a streamLength
+         * and \a totalSamples.
+         *
+         * \note totalSamples can be zero if the file doesn't have a 'fact' chunk.
+         */
+        Properties(const ByteVector &data, uint streamLength, uint totalSamples, ReadStyle style);
```

Comment:
Looking at this more closely, another option is creating a thin wrapper type that we require usage of here, e.g.:

    class SampleLength
    {
    public:
      explicit SampleLength(int length);
      operator int() const;
      // ...
    };

That would probably make sense if this is a pattern we see across most of the file types.